libflux/core/src/ast/mod.rs

```diff
@@ -338,8 +329,8 @@ pub struct BaseNode {
     // use the base node comments when a non-terminal contains just one
     // terminal on the right hand side. This saves us populating the
     // type-specific AST nodes with comment lists when we can avoid it..
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub comments: CommentList,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub comments: Vec<String>,
```

Comment:
Can we keep the `Comment` struct and make this a vector of comments? I'd like to keep the struct so that we can add additional information to the node in the future if we want it. For example, I don't know if we want to put position information in with the comments at some point. If we do, then using a `Comment` struct keeps that possibility open. If we use a vector of strings, that becomes impossible.

Something like:

```rust
pub struct Comment {
    text: String,
}

pub struct BaseNode {
    pub comments: Vec<Comment>,
}
```

Reply:
That has been implemented too.